clang/include/clang/Frontend/SerializedDiagnosticPrinter.h

```diff
@@ -31,9 +31,12 @@ namespace serialized_diags {
 /// This allows wrapper tools for Clang to get diagnostics from Clang
 /// (via libclang) without needing to parse Clang's command line output.
 ///
-std::unique_ptr<DiagnosticConsumer> create(StringRef OutputFile,
-                                           DiagnosticOptions *Diags,
-                                           bool MergeChildRecords = false);
+/// \param OS optional stream to output the serialized diagnostics buffer,
+/// instead of writing out directly to a file.
+std::unique_ptr<DiagnosticConsumer>
+create(StringRef OutputFile, DiagnosticOptions *Diags,
```

Comment:
Trying to understand what is the condition that we need to write to file and when we need to write to OS? Will be good have a comment to describe that.

Also should this just take OS as parameter and drop OutputFile? If we ever need to write to both, we can just create a mirroring output backend?

Reply:
Even with `OS`, the `OutputFile` is still useful to pass if you enable `MergeChildRecords`, in which case the implementation would open the existing file to read the diagnostics and merge them.

Overall, I consider these changes in `SerializedDiagnosticPrinter` as temporary transition until we have structured diagnostics caching as I mention in [this FIXME](https://github.com/apple/llvm-project/pull/5078/commits/58f2dd0701e19fa9a28480b2531a66154420763c#diff-c6f8f185d25be2e8d40d9c42ffcca1caac9b2f8403da61bd7957b6957763cb86R423), after which we'll stop overriding the caller's `DiagnosticConsumer` and making assumptions about how it should be setup. We won't need to manage serialized diagnostics files explicitly and I would revert these changes to `SerializedDiagnosticPrinter` (which is also the reason I didn't want to introduce them upstream).

I'll add a comment to clarify this aspect.

Reply:
Added a FIXME comment along in the doc-comments here.